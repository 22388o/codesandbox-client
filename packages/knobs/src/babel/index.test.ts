import * as core from '@babel/core';
import babelPlugin from './index';

function validateBabel(code: string) {
  // @ts-ignore
  const { code: outputCode } = core.transformSync(code, {
    plugins: [babelPlugin],
    filename: __dirname + '/test.js',
  });

  expect(outputCode).toMatchSnapshot();
}

it('converts knobs to something new', () => {
  const code = `import * as knobs from 'knobs';
  
  let var1 = knobs.number('padding');
  let var2 = knobs.string('name');
  `;

  validateBabel(code);
});
