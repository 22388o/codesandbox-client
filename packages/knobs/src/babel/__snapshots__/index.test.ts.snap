// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts knobs to something new 1`] = `
"import * as _testJsKnobsJson from \\"./test.js.knobs.json\\";
import * as knobs from 'knobs';
let var1 = knobs.number(_testJsKnobsJson.padding, 'padding');
let var2 = knobs.string(_testJsKnobsJson.name, 'name');"
`;
